test/e2e/helpers.js

```diff
@@ -1008,7 +1008,7 @@ async function validateContractDetails(driver) {
     '.signature-request-content__verify-contract-details',
   );
 
-  verifyContractDetailsButton.click();
+  await verifyContractDetailsButton.click();
```

Comment:
hi there! That's a really nice finding!! I was wondering, could we, instead of using the `click` use the `driver.clickElement()`?


This way, we'll benefit from the guards inside the function. As it is now, it can introduce a race condition where the element is not enabled/clickable:

https://github.com/MetaMask/metamask-extension/blob/develop/test/e2e/webdriver/driver.js#L568

Reply:
nicee, sounds good to me! 

I created another PR for the fix with your suggestion here https://github.com/MetaMask/metamask-extension/pull/26040

going to revert this fix in this PR to keep the logic separate

Reply:
thanks for the suggestion! reverted this line here https://github.com/MetaMask/metamask-extension/pull/25769/commits/b86b30b02e41541ec4201bf045d43be4c54c5243 to address flaky "Signature Approved Event" in PR mentioned above

Reply:
awesome, thank you !! I've already approved it :heavy_check_mark: